src/emit.ts

```diff
@@ -64,4 +65,19 @@ export const recordEvent = (
 
   // Record the event message sent by the emit
   events[cid][event].push(args)
+
+  // see if we are emitting a v-model syntax sugar update:prop event
+  const match = event.match(/^(update:)(.*)/)
+
+  if (match && match[1] === 'update:' && vModelProps.includes(match[2])) {
+    if (args.length !== 1) {
+      throw new Error(
```

Comment:
Missing a test for this scenario? Not that common, but it would be great to know we don't run into some regression